src/typing_manager.ts

```diff
@@ -72,7 +72,7 @@ export class TypingManager implements Disposable {
                 );
                 this.changeManager.getDocumentChangeCompletionLock(editor.document)?.then(() => {
                     this.isEnteringInsertMode = false;
-                    if (this.typeHandlerDisposable) {
+                    if (this.typeHandlerDisposable && this.modeManager.isInsertMode) {
                         this.logger.debug(`${LOG_PREFIX}: Waiting done, disposing type handler`);
                         this.typeHandlerDisposable.dispose();
```

Comment:
better to wrap these lines into a check instead because you might lose the pressed keys otherwise

Or move `if(this.pendingKeysAfterEnter)` check outside of this block

Reply:
Thank you, I just wanted to ask about this :)

Reply:
Done.